src/RA_Dlg_AchievementsReporter.cpp

```diff
@@ -126,40 +133,30 @@ INT_PTR CALLBACK Dlg_AchievementsReporter::AchievementsReporterProc(HWND hDlg, U
             {
                 case IDOK:
                 {
-                    HWND hList = GetDlgItem(hDlg, IDC_RA_REPORTBROKENACHIEVEMENTSLIST);
-
-                    const bool bProblem1Sel = (IsDlgButtonChecked(hDlg, IDC_RA_PROBLEMTYPE1) == BST_CHECKED);
-                    const bool bProblem2Sel = (IsDlgButtonChecked(hDlg, IDC_RA_PROBLEMTYPE2) == BST_CHECKED);
+                    const auto hList{ ::GetDlgItem(hDlg, IDC_RA_REPORTBROKENACHIEVEMENTSLIST) };
+                    
+                    const auto bProblem1Sel{ Button_GetCheck(GetDlgItem(hDlg, IDC_RA_PROBLEMTYPE1)) };
+                    const auto bProblem2Sel{ Button_GetCheck(GetDlgItem(hDlg, IDC_RA_PROBLEMTYPE2)) };
 
                     if ((bProblem1Sel == false) && (bProblem2Sel == false))
                     {
                         MessageBox(nullptr, TEXT("Please select a problem type."), TEXT("Warning"), MB_ICONWARNING);
                         return FALSE;
                     }
 
-                    const int nProblemType = bProblem1Sel ? 1 : bProblem2Sel ? 2 : 0;	// 0==?
-                    const char* sProblemTypeNice = PROBLEM_STR[nProblemType];
-
-                    char sBuggedIDs[1024];
-                    sprintf_s(sBuggedIDs, 1024, "");
+                    std::string sBuggedIDs;
 
                     int nReportCount = 0;
-
-                    const size_t nListSize = ListView_GetItemCount(hList);
-                    for (size_t i = 0; i < nListSize; ++i)
+                    const int nListSize = ListView_GetItemCount(hList);
+                    for (auto i = 0; i < nListSize; ++i)
                     {
                         if (ListView_GetCheckState(hList, i) != 0)
                         {
-                            //	NASTY big assumption here...
-                            char buffer[1024];
-                            sprintf_s(buffer, 1024, "%u,", g_pActiveAchievements->GetAchievement(i).ID());
-                            strcat_s(sBuggedIDs, 1024, buffer);
-
-                            //ListView_GetItem( hList );	
+                            sBuggedIDs.append(ra::StringPrintf("%zu, ", g_pActiveAchievements->GetAchievement(i).ID()));
```

Comment:
You've added an extra space here. While it might be visually appealing, this also gets sent to the API and may have unintended side effects there. For now, recommend removing the space.